---
author: Boddu Jyothika
output:
  html_document:
    toc: true
    toc_depth: 4
    toc_float: true
    dev: 'svg'
  md_document:
    variant: gfm
bibliography: bibliography.ris
nocite: '@*'
---
## Variant Calling
Variant naming is the recognition of genetic differences between individuals of the same species (germline) or tissue differences in a single (somatic) organism.

First move is to use the wget command from the path specified in the module to get the reference genome by using -c, output by-O and we need to gunzip the script.
The second step is to re-read the NGS by using the fastqdump command of the SRA toolkit and to splitfiles so that we get the two SRA files under SRR6808334 1.fastq and SRR6808334 2.fastq accession.
Third step is to trim the reads that we will trim the reads using the trimmomatic in to paired and unpaired java system with two files that are paired and two unpaired resulting in a total of four data. We'll also construct log files, and err files.
Fourth step is to index the genome for the reference file using the Bwa index BWA is a software package for mapping DNA sequences against a large reference genome, like the human genome.
It consists of three algorithms: BWA-backtrack, BWA-SW and BWA-MEM and we obtain outfiles such as.sa,.amb,.amb,.pac,.bwt which are used to test the alignments and also samtool faidx which index the files and generate an output of.fai file indexing the fasta format reference sequences.
Fifth stage is to align the reads using Bwa mem reads, this method aligns the single end reads or paired end reads to a chosen reference genome using BWA MEM Algorithm which reads to be distributed in the fastq format from which we can obtain the.sam file performance.
Sixth step is Sorting the file generated by bwa mem to a sorted bam by using samtools sort to get a sorted type of sam file that is the compressed binary version of sam named the.sorted.bam file.
This is also done by using the samtools index, which indexes the.sorted.bam file as the input and generates the.sorted.bam.bai file as the output by indexing the reads as the output
he last step is the Produce a VCF file using DeepVariant DeepVariant is an analysis pipeline that uses a deep neural network to call genetic variants from next-generation DNA sequencing data. DeepVariant relies on Nucleus, a library of Python and C++ code for reading and writing data in common genomics file formats (like SAM and VCF) designed for painless integration with the TensorFlow machine learning framework.
##References
